src/poseidon/poseidon-opt.ts

```diff
@@ -65,5 +67,50 @@ export class Poseidon {
 
     return F.normalize(state[0]);
   }
+
+  hashBytes(msg: Uint8Array): bigint {
+    let inputs: Array<bigint> = [];
+    for (let i = 0; i < SPONGE_INPUTS; i += 1) {
+      inputs.push(BigInt(0));
+    }
+    let dirty = false;
+    let hash: BigInt;
+
+    let k = 0;
+    for (let i = 0; i < parseInt(`${msg.length / SPONGE_CHUNK_SIZE}`); i += 1) {
+      dirty = true;
+      inputs[k] = utils.beBuff2int(msg.slice(SPONGE_CHUNK_SIZE * i, SPONGE_CHUNK_SIZE * (i + 1)));
+      if (k === SPONGE_INPUTS - 1) {
+        hash = this.hash(inputs);
+        dirty = false;
+        inputs = [];
+        inputs[0] = hash.valueOf();
+        for (let j = 1; j < SPONGE_INPUTS; j += 1) {
+          inputs[j] = BigInt(0);
+        }
+        k = 1;
+      } else {
+        k += 1;
+      }
+    }
+
+    if (msg.length % SPONGE_CHUNK_SIZE != 0) {
+      let buff = new Uint8Array(new ArrayBuffer(SPONGE_CHUNK_SIZE));
```

Comment:
new Uint8Array(SPONGE_CHUNK_SIZE) ?

Reply:
Amazing,  just checked the docs and wasn't aware of this constructor variant, always thought you need an ArrayBuffer or Array to construct them :)

Changed in [this](https://github.com/iden3/js-crypto/pull/2/commits/5cc7f47b45c5d8f19ab6092311b9bcb34e30642a) commit.